kubernetes/cache/istio.go

```diff
@@ -761,6 +786,128 @@ func (c *kialiCacheImpl) GetTelemetries(namespace, labelSelector string) ([]*v1a
 	return []*v1alpha1.Telemetry{}, nil
 }
 
+func (c *kialiCacheImpl) GetK8sGateway(namespace, name string) (*gatewayapi.Gateway, error) {
+	if !c.CheckIstioResource(kubernetes.K8sGateways) {
+		return nil, fmt.Errorf("Kiali cache doesn't support [resourceType: %s]", kubernetes.K8sGatewayType)
+	}
+	if nsCache, ok := c.nsCache[namespace]; ok {
+		// Cache stores natively items with namespace/name pattern, we can skip the Indexer by name and make a direct call
+		key := namespace + "/" + name
+		obj, exist, err := nsCache[kubernetes.K8sGatewayType].GetStore().GetByKey(key)
+		if err != nil {
+			return nil, err
+		}
+		if exist {
+			l, ok := obj.(*gatewayapi.Gateway)
+			if !ok {
+				return nil, errors.New("bad K8sGateway type found in cache")
+			}
+			l.Kind = kubernetes.K8sGatewayType
```

Comment:
This is a bit of a problem since the actual Kind is `Gateway` and not `K8sGateway`. This is more of a problem with how Kiali handles Kind types in general. Kiali assumes that the Kind(s) are unique but as seen here they are not. The GVKs _are_ unique (Group + Version + Kind) but everywhere in Kiali we assume that the Kinds alone are unique including in the UI.

From the Istio Config page:
![Screenshot from 2022-09-14 16-15-50](https://user-images.githubusercontent.com/6226732/190253931-f5ed0551-0dc9-4a4b-8dfc-7a0f8a2b436a.png)

`K8sGateway` is not an actual Kind but we need some way of distinguishing it from the `networking.istio.io` `Gateway`. I'm not entirely sure how users perceive what the `Type` column represents there but I'm leery of setting this to a random `Kind` that doesn't actually exist.

The Istio config editor is a good example of this:
![Screenshot from 2022-09-14 13-53-31](https://user-images.githubusercontent.com/6226732/190254937-ead69a77-379f-49b5-94c5-3be6f86ccaf3.png)

Compare this to the output of `kubectl`:
```
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"gateway.networking.k8s.io/v1alpha2","kind":"Gateway","metadata":{"annotations":{},"name":"gateway","namespace":"istio-ingress"},"spec":{"gatewayClassName":"istio","listeners":[{"allowedRoutes":{"namespaces":{"from":"All"}},"hostname":"*.example.com","name":"default","port":80,"protocol":"HTTP"}]}}
  creationTimestamp: "2022-09-14T20:47:04Z"
  generation: 1
  name: gateway
  namespace: istio-ingress
  resourceVersion: "4080"
```

It's confusing to have Kiali tell you one thing but `kubectl` and other tooling tell you another. Not sure what a good solution is here without doing a bunch of refactoring to include GVK in places where we always assume Kind. Perhaps we can display the correct `Gateway` kind in the istio config yaml editor at least?

Reply:
Yeah there is  a confusion between two Gateways on Kiali backend, also on UI side.
I provided a solution to differentiate two types of objects.
That YAML editor value was hard-coded to show G8sGateway then it changes to Gateway while pushing.
But at least I can change how it is displayed in UI. Then question is the filters, which now have K8s Types of filters..